docs/start/data/custom.md

```diff
@@ -64,13 +64,13 @@ Routes can take most of their definition lazily with the `lazy` property.
 createBrowserRouter([
   {
     path: "/show/:showId",
-    lazy: () => {
-      let [loader, action, Component] = await Promise.all([
-        import("./show.action.js"),
-        import("./show.loader.js"),
-        import("./show.component.js"),
-      ]);
-      return { loader, action, Component };
+    lazy: {
+      loader: async () =>
+        (await import("./show.loader.js")).loader,
+      action: async () =>
+        (await import("./show.action.js")).action,
+      Component: async () =>
+        (await import("./show.component.js")).Component,
```

Comment:
huh - I didn't realize our docs showed this type of granular file splitting.  I feel like the more common use case is `lazy: () => import("./route")` so I would actually vote to document that version with a note on how to get more granular if/when needed

Reply:
I feel this is the better API to document because it handles all properties, whereas the lazy function can't handle middleware. I also feel like this version is a bit clearer in terms of what's going on.